File: backend_flask/routes/profile_routes.py
from controllers.profile_controller import (
    get_profile,
    update_profile,
    upload_profile_avatar,
)
from middleware.auth_middleware import require_auth


def register_profile_routes(app):
    app.get("/api/profile")(require_auth(get_profile))
    app.post("/api/profile/avatar")(require_auth(upload_profile_avatar))
    app.patch("/api/profile")(require_auth(update_profile))

File: backend/src/routes/profile.routes.js
import express from "express";
import {
  getProfile,
  getProfileSubmissions,
  uploadProfileAvatar,
  updateProfile,
} from "../controllers/profile.controller.js";
import { requireAuth } from "../middleware/auth.middleware.js";

const router = express.Router();

router.use(requireAuth);

router.get("/submissions", getProfileSubmissions);
router.get("/", getProfile);
router.post("/avatar", uploadProfileAvatar);
router.patch("/", updateProfile);

export default router;
